runtime/src/set_auth_manifest.rs

```diff
@@ -401,6 +415,45 @@ impl SetAuthManifestCmd {
             soc_ifc,
         )?;
 
+        // Sort the image metadata list by firmware ID in place. Also check for duplicate firmware IDs.        let slice =
+        let slice =
+            &mut metadata_mailbox.image_metadata_list[..metadata_mailbox.entry_count as usize];
+
+        Self::sort_and_check_duplicate_fwid(slice)?;
```

Comment:
This seems unnecessarily complicated. I can't imagine linear search being noticeably slow here.

Reply:
Given we had 128 (now 127) hashes, there was some concern about the search runtimes.